src/rocks_engine.cpp

```diff
@@ -352,18 +353,18 @@ namespace mongo {
         auto config = _getIdentConfig(ident);
         std::string prefix = _extractPrefix(config);
 
-        RocksRecordStore* recordStore =
+        std::unique_ptr<RocksRecordStore> recordStore =
             options.capped
-                ? new RocksRecordStore(
+                ? stdx::make_unique<RocksRecordStore>(
                       ns, ident, _db.get(), _counterManager.get(), prefix, true,
                       options.cappedSize ? options.cappedSize : 4096,  // default size
                       options.cappedMaxDocs ? options.cappedMaxDocs : -1)
-                : new RocksRecordStore(ns, ident, _db.get(), _counterManager.get(),
+                : stdx::make_unique<RocksRecordStore>(ns, ident, _db.get(), _counterManager.get(),
                                        prefix);
 
         {
             stdx::lock_guard<stdx::mutex> lk(_identObjectMapMutex);
-            _identCollectionMap[ident] = recordStore;
+            _identCollectionMap[ident] = recordStore.get();
```

Comment:
I do think this leaves extra entries with dangling pointers in the _identCollectionMap. However, these pointers should not be accessible anymore, as the ident won't be looked up again, so it just wastes a little bit of space.
No ownership semantics are changes as part of this patch.